application/ui/components/param/text.jsx

```diff
@@ -0,0 +1,34 @@
+/** @jsx React.DOM */
+'use strict';
+
+var React = require('react');
+
+module.exports = React.createClass({
+
+    displayName : 'TextParam',
+
+    propTypes : {
+        value    : React.PropTypes.oneOfType([
+            React.PropTypes.string,
+            React.PropTypes.number
+        ]),
+        type     : React.PropTypes.string,
+        onChange : React.PropTypes.func
+    },
+
+    handleChange : function(event)
+    {
+        var value = event.target.value;
+        if (this.props.type === 'number') {
+            value = parseInt(value);
```

Comment:
Make sure to provide a radix of `10` as the second argument of `parseInt`. (JSHint normally complains about this.)

> Different implementations produce different results when a radix is not specified.
> ~ [parseInt on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt)